src/NuGet.Services.Metadata.Catalog.Monitoring/Status/PackageMonitoringStatusService.cs

```diff
@@ -0,0 +1,171 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using NuGet.Protocol;
+using NuGet.Services.Metadata.Catalog.Helpers;
+using NuGet.Services.Metadata.Catalog.Persistence;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring
+{
+    /// <summary>
+    /// Manages the storage and access of the status of packages that validation has run against.
+    /// </summary>
+    public class PackageMonitoringStatusService : IPackageMonitoringStatusService
+    {
+        private ILogger<PackageMonitoringStatusService> _logger;
+
+        /// <summary>
+        /// The <see cref="StorageFactory"/> to use to save status of packages.
+        /// </summary>
+        private StorageFactory _storageFactory;
+
+        /// <summary>
+        /// The <see cref="JsonSerializerSettings"/> to use to save status of packages.
+        /// </summary>
+        public JsonSerializerSettings SerializerSettings => _serializerSettings.Value;
+        private Lazy<JsonSerializerSettings> _serializerSettings = new Lazy<JsonSerializerSettings>(() =>
+        {
+            var settings = new JsonSerializerSettings();
+
+            settings.Converters.Add(new NuGetVersionConverter());
+            settings.Converters.Add(new StringEnumConverter());
+
+            return settings;
+        });
+
+        public PackageMonitoringStatusService(StorageFactory storageFactory, ILogger<PackageMonitoringStatusService> logger)
+        {
+            _logger = logger;
+            _storageFactory = storageFactory;
+        }
+
+        public Task<PackageMonitoringStatus> Get(FeedPackageIdentity package, CancellationToken token)
+        {
+            foreach (var state in Enum.GetNames(typeof(PackageMonitoringStatus.PackageState)))
```

Comment:
This will attempt to get `Valid` states before `Invalid` states as that is the order the enum's values are listed, correct? Would it be better to have `Invalid` be first so that `Invalid` package statuses are favored?